modules/backend/models/User.php

```diff
@@ -209,4 +209,56 @@ public function unsuspend()
     {
         BackendAuth::findThrottleByUserId($this->id)->unsuspend();
     }
+
+    //
+    // Impersonation
+    //
+
+    /**
+     * Returns an array of merged permissions based on the user's individual permissions
+     * and their group permissions filtering out any permissions the impersonator doesn't
+     * have access to (if the current user is being impersonated)
+     *
+     * @return array
+     */
+    public function getMergedPermissions()
+    {
+        if (!$this->mergedPermissions) {
```

Comment:
@LukeTowers we'll need to make sure that the merged permissions of an impersonatee are cleared immediately on impersonation, on the off-chance that some functionality is fired immediately after impersonation, before the model has had a chance to refresh on, say, a page refresh.

Reply:
Good catch!